### ISSUE TITLE
Pyright/Pylance does not recognize dynamic attributes added to QApplication subclasses (PyQt/PySide)

### ISSUE THREAD

**Issue by crazy-italian-coder** [microsoft/pyright#11352]
Hi,  
in PyQt and PySide projects, it is common to subclass `QApplication` and add custom attributes (e.g., logger, translator, error codes) at runtime.  
However, Pyright/Pylance always reports `"Attribute ... is unknown"` when accessing these attributes via `QApplication.instance()`, even if they are set in the subclass’ `__init__`.

Example:
```python
from PyQt5.QtWidgets import QApplication

class MyApp(QApplication):
    def __init__(self, argv):
        super().__init__(argv)
        self.logger = "my logger"

app = MyApp([])
logger = QApplication.instance().logger  # Pyright: "Attribute 'logger' is unknown"
```

This is valid and idiomatic PyQt code, but the static analyzer cannot see the dynamic attribute.  
Disabling `"reportAttributeAccessIssue"` works, but disables useful checks globally.

**Expected:**  
Pyright/Pylance should allow a way to mark dynamic attributes as valid for a given class, or at least not report errors when accessing attributes set in a known subclass.

Thank you!

**Comment by rchiodo** [comment]
Sorry but this is by design at the moment. Pyright is a static analysis tool. It can't know what type `QApplication.instance()` is going to return. 

To work around this, you'd need a cast:

```python
from typing import cast

from PyQt5.QtWidgets import QApplication

class MyApp(QApplication):
    def __init__(self, argv):
        super().__init__(argv)
        self.logger = "my logger"

app = MyApp([])
instance : MyApp = cast(MyApp, QApplication.instance())
logger = instance.logger
```
